### PR TITLE
graphio: add function to guess file format

### PR DESCRIPTION
**Pull request by bernlu**
This PR adds a new `guessFileFormat` function to the `graphio` module (python). This function is used to guess the file format when `graphio.readGraph(path, fileformat)` is called with `fileformat=None` - which is now the default value for the `fileformat` parameter. Guessing may fail and raise an exception (in which case one needs to provide the format manually as before). 

The guessing strategies are quite simple and will probably fail for some cases where the format is actually rather easy to guess, but for now it works for the most common file structures and types (see tests). We can improve this function when new ideas/problems arise.

Direct use:
```py
>>> import networkit as nk
>>> nk.graphio.guessFileFormat("input/airfoil1.graph")
<Format.METIS: 6>
```
With `readGraph`:
```py
>>> import networkit as nk
>>> nk.readGraph("input/airfoil1.graph")
<networkit.graph.Graph object at 0x7fa2a591c570>
```

One problem which is unfixable is that a file can be both in METIS and EdgeList format, e.g.
```
3 3
1 2
0 2
0 1
```
Depending on the format, this is either a 3 or 4 node graph. When this case is detected, an error is raised and the format has to be provided manually.

### CODE REVIEW

**Comment by bernlu** [comment]
Maybe it would be nice to add support for MatrixMarket files as well, but I think currently the matrix market file reader is c++ only.. which means this would be a larger change and probably its own PR.

**Comment by bernlu** [comment]
Added python support for reading matrix market files to graphs. this does not use the cpp matrix market reader because that reader creates a CSRMatrix (which is not what we need for readGraph). I decided to instead just read the matrix market file as edge list.

**Review by fabratu — COMMENTED** [review]
Only two minor comments concerning documentation - otherwise, I think we can merge this. As discussed offline, the `MatrixMarket` reader for Cython/Python (returning a graph, not a `CSRMatrix`) is added in another PR.

File: networkit/graphio.pyx [hunk 1]
```diff
@@ -1098,17 +1101,161 @@ def getReader(fileformat, *kargs, **kwargs):
...
+	with open(filepath, 'r') as f:
+		n = None
+		m = None
+		noncommentlines = 0
+		edges = 0
+		for line in f.readlines():
+			# snap check: keep verifying line format while the file could be a snap file
+			# check that all lines have format \d+ \d+ (or are empty lines)
+			if snapFound:
+					match = re.match(r'(^\d+\s\d+$)|(^\s*$)', line)
```
Comment (bernlu) [inline networkit/graphio.pyx:1187]: snap format: lines may end with whitespaces

**Review by fabratu — APPROVED** [review]

**Pull request merged**